src/main/resources/templates/jenkins/java/blackbox/regularRuns/pipeline.groovy

```diff
@@ -0,0 +1,281 @@
+        docker.image(dockerImage).inside("-v ${secretTestVolume}:${secret_volume_mount_path}") { c ->
+            sh """
+            rm -rf ${secret_volume_mount_path}/*
+            cp testsuite/${tool}.tests/secret.exp ${secret_volume_mount_path}/secret.exp
+            """
+
+            if (hasSecretTestFiles) {
+                sh("cp -vr ${testfiles_base_path}/secret ${secret_volume_mount_path}/secret")
+            }
+        }
+    }
+}
+
+/**
+ * Builds the student code and tries to find a main method.
+ */
+private void build() {
+    stage('Build') {
+        docker.image(dockerImage).inside(dockerFlags) { c ->
+            sh "mvn ${mavenFlags} clean compile"
+            setMainClass()
+        }
+    }
+}
+
+/**
+ * Runs the custom {@code pipeline-helper.jar} checkers.
+ */
+private void customCheckers() {
+    stage('Checkers') {
+        docker.image(dockerImage).inside(dockerFlags) { c ->
+            // all java files in the assignment folder should have maximal line length 80
+            sh 'pipeline-helper -l 80 assignment/ java'
+            // checks that the file exists and is not empty for non gui programs
+            if (runDejagnuTests) {
+                sh 'pipeline-helper -e assignment/Tests.txt'
+            }
+        }
+    }
+}
+
+/**
+ * Runs Dejagnu scripts.
+ */
+private void dejagnuTests() {
+    stage('Public+Advanced Tests') {
+        docker.image(dockerImage).inside(dockerFlags) { c ->
+            applyExpectScriptReplacements()
+
+            runDejagnuTestStep("public", 60)
+            runDejagnuTestStep("advanced", 60)
+        }
+    }
+
+    stage('Secret Tests') {
+        docker.image(dockerImage).inside(dockerFlags + " -v ${secretTestVolume}:${secret_volume_mount_path}:ro") { c ->
+            restoreSecretFiles()
+            applyExpectScriptReplacements()
+            runDejagnuTestStep("secret", 60)
+        }
+    }
+}
+
+/**
+ * Runs a single Dejagnu test step.
+ *
+ * @param step The test step name. Runs the test file {@code step.exp}.
+ * @param timeoutSeconds A number of seconds after which Jenkins will kill the dejagnu process.
+ */
+private void runDejagnuTestStep(String step, int timeoutSeconds) {
+    catchError() {
+        timeout(time: timeoutSeconds, unit: 'SECONDS') {
+            sh """
+            cd testsuite || exit
+            rm ${tool}.log || true
+            runtest --tool ${tool} ${step}.exp || true
+            """
+        }
+    }
+    sh("""pipeline-helper -o customFeedbacks -d "dejagnu[${step}]" testsuite/${tool}.log""")
+}
+
+/**
+ * Extracts the Dejagnu tool name from the folder names.
+ * <p>
+ * E.g., for a folder {@code testsuite/gcd.tests/} the tool name is {@code gcd}.
+ *
+ * @return The Dejagnu tool name.
+ */
+private String getToolName() {
+    // Java Files API gets blocked by Jenkins sandbox
+
+    return sh(
+        script: """find testsuite -name "*.tests" -type d -printf "%f" | cut --delimiter=. -f 1""",
```

Comment:
In case the package name contains multiple dots the toolname detection might break which might have caused the issue @laadvo mentioned.

Probable fix: use `sed` to remove the `.tests` at the end
```suggestion
        script: """find testsuite -name "*.tests" -type d -printf "%f" | sed 's#.tests$##'""",
```

Reply:
I think this might be the issue. I used a package name with multiple dots when testing